packages/snaps-controllers/src/interface/SnapInterfaceController.ts

```diff
@@ -299,7 +323,11 @@ export class SnapInterfaceController extends BaseController<
     validateInterfaceContext(context);
 
     const oldState = this.state.interfaces[id].state;
-    const newState = constructState(oldState, element);
+    const newState = constructState(oldState, element, {
+      getAssetMetadata: this.#getAssetMetadata.bind(this),
```

Comment:
Not sure adding an extra hook that accesses the same state as `getAssetsState` is worth it here, would probably just use `getAssetsState` everywhere